src/main/java/com/limechain/trie/decoder/TrieLeafDecoder.java

```diff
@@ -1,26 +1,46 @@
 package com.limechain.trie.decoder;
 
 import com.limechain.trie.Node;
+import com.limechain.trie.NodeVariant;
 import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.types.Hash256;
+import lombok.experimental.UtilityClass;
 
+/**
+ * This class is used to decode leaf nodes from a ScaleCodecReader input stream.
+ */
+@UtilityClass
 public class TrieLeafDecoder {
     /**
-     * Decodes a leaf node from a ScaleCodecReader input stream.
+     * Decodes a leaf or hashed leaf node from a ScaleCodecReader input stream.
      *
-     * @param reader the ScaleCodecReader to read the encoded node data from
+     * @param reader           the ScaleCodecReader to read the encoded node data from
+     * @param variant          the variant of the node to be decoded
      * @param partialKeyLength the length of the partial key to be read
      * @return the decoded Node object
      * @throws TrieDecoderException if an error occurs while decoding the node. This could be
-     * due to an issue reading the children bitmap or the storage value.
+     *                              due to an issue reading the children bitmap or the storage value.
```

Comment:
Is this indentation intentional?

Reply:
Yes, it's meant to be aligned with the upper line
<img width="817" alt="image" src="https://github.com/LimeChain/Fruzhin/assets/29047760/dae3d8d3-f52f-425d-a583-4426148a0b1c">